app/workers/builds.php

```diff
@@ -179,6 +188,7 @@ protected function buildDeployment(Document $project, Document $function, Docume
             if ($deployment->getAttribute('activate') === true) {
                 $function->setAttribute('deployment', $deployment->getId());
                 $function = $dbForProject->updateDocument('functions', $function->getId(), $function);
+                $dbForProject->deleteCachedDocument('functions', $function->getId());
```

Comment:
I think we should delete the cache only when we update a variable. When updating the function itself the cache should be automatically cleaned.

Reply:
Please recheck all usage of this method in the PR.